config/config.go

```diff
@@ -136,3 +138,31 @@ func Parse(raw map[string]string) (Config, error) {
 
 	return config, nil
 }
+
+// GetOptions returns generated options for mongo connection depending on mechanism.
+func (d *Config) GetOptions() *options.ClientOptions {
```

Comment:
Since we don't have a lot of logic in the `GetOptions` and `getURIAndPropertiesByMechanism` methods, we can probably combine them into one method.

I suggest doing something like this, but this is open for debate:
```go
// GetClientOptions returns options to configure a MongoDB client instance based on the underlying [Config].
func (d *Config) GetClientOptions() *options.ClientOptions {
	opts := options.Client().ApplyURI(d.URI)

	// if some of the auth properties are set, we need to perform an authentication
	if d.Auth != (AuthConfig{}) {
		auth := options.Credential{
			AuthMechanism: string(d.Auth.Mechanism),
			AuthSource:    d.Auth.DB,
			Username:      d.Auth.Username,
			Password:      d.Auth.Password,
		}

		// set AWS session token if it's not empty
		if d.Auth.AWSSessionToken != "" {
			auth.AuthMechanismProperties = map[string]string{
				awsSessionTokenPropertyName: d.Auth.AWSSessionToken,
			}
		}

		// if the authentication mechanism is X509,
		// we need to specify file paths pointed to TLS certificates in the URI
		if d.Auth.Mechanism == X509 {
			// ignore the error here, because we have a validation in the Parse function,
			// and we sure that the d.URI is a valid URI
			// TODO: change the d.URI field type from string to *url.URL
			parsedURI, _ := url.Parse(d.URI)
			values := parsedURI.Query()
			values.Add(tlsCAFileQueryName, d.Auth.TLSCAFile)
			values.Add(tlsCertificateKeyFileQueryName, d.Auth.TLSCertificateKeyFile)
			parsedURI.RawQuery = values.Encode()
			opts = opts.ApplyURI(parsedURI.String())
		}

		opts = opts.SetAuth(auth)
	}

	return opts
}
```

I left a note here saying that it will be better to have the `d.URI` field as a `*url.URL`. I will change this field in a separate pull request, so don't bother yourself about that.

Reply:
Your version also looks good to me. Let's just add the following comment under `parsedURI, _ := url.Parse(d.URI)`:
```go
// ignore the error here, because we have a validation in the Parse function,
// and we sure that the d.URI is a valid URI
// TODO: change the d.URI field type from string to *url.URL
```

In order to not forget to change the type of the `d.URI` field in the future.